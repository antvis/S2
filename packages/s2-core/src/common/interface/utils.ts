export type PickEssential<O> = {
  [K in keyof O as Pick<Partial<O>, K> extends Pick<O, K> ? never : K]: O[K];
};
