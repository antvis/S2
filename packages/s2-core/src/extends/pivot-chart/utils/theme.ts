import {
  SpreadSheet,
  getColCellTheme,
  getCornerCellTheme,
  getRowCellTheme,
  type S2Theme,
  type SimplePalette,
} from '@antv/s2';
import { merge } from 'lodash';
import { AxisCellType } from '../cell/cell-type';

export const getCustomTheme = (
  palette: SimplePalette,
  spreadsheet?: SpreadSheet,
): S2Theme => {
  return {
    [AxisCellType.AXIS_CORNER_CELL]: getCornerCellTheme(palette),
    [AxisCellType.AXIS_ROW_CELL]: getRowCellTheme(palette, spreadsheet),
    [AxisCellType.AXIS_COL_CELL]: merge(getColCellTheme(palette), {
      measureText: {
        textAlign: 'center',
      },
    }),
  };
};
