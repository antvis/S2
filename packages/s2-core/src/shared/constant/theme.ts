import { S2_PREFIX_CLS } from '../../common';

/**
 * 注入主题 css 变量的节点名
 */
export const STYLE_ELEMENT_ID = `${S2_PREFIX_CLS}-component-vars`;
