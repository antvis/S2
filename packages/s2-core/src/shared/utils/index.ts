export * from './classnames';
export * from './drill-down';
export * from './options';
export * from './resize';
