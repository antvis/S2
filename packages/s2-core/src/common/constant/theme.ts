import { Palette } from '../interface';
import { paletteColorful } from '@/theme/palette/colorful';
import { paletteDefault } from '@/theme/palette/default';
import { paletteGray } from '@/theme/palette/gray';

// Map of the theme
export const PALETTE_MAP: Record<string, Palette> = {
  default: paletteDefault,
  colorful: paletteColorful,
  gray: paletteGray,
};

export const FONT_FAMILY =
  'Roboto, PingFangSC, BlinkMacSystemFont, Microsoft YaHei, Arial, sans-serif';

export const MINI_BAR_CHART_HEIGHT = 12;
