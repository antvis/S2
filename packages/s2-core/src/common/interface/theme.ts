import { ShapeAttrs } from '@antv/g-canvas';
import { InteractionStateName } from '../constant';
import { CellTypes } from '@/common/constant/interaction';

// 文本内容的水平对齐方式, 默认 left
export type TextAlign = 'left' | 'center' | 'right';

// 绘制文本时的基线, 对应垂直方向对齐方式 默认 bottom
export type TextBaseline = 'top' | 'middle' | 'bottom';

export interface Palette {
  /* 基础色值 */
  basicColors: string[];
  /* 语义色值 */
  semanticColors: {
    red?: string;
    green?: string;
    /* 额外颜色字段 */
    [key: string]: string;
  };
}

export interface Padding {
  top?: number;
  right?: number;
  bottom?: number;
  left?: number;
}

export interface Background {
  opacity?: number;
  color?: string;
}
export interface InteractionStateTheme {
  /* 背景透明度 */
  backgroundOpacity?: number;
  /* 背景填充色 */
  backgroundColor?: string;
  /* 文本透明度 */
  textOpacity?: number;
  /* 边线颜色 */
  borderColor?: string;
  /* 边线宽度 */
  borderWidth?: number;
  /* 边线透明度 */
  borderOpacity?: number;
  /* 透明度 */
  opacity?: number;
}

export type InteractionState = {
  [K in InteractionStateName]?: InteractionStateTheme;
};

export type Margin = Padding;

export interface TextAlignCfg {
  textAlign?: TextAlign;
  textBaseline?: TextBaseline;
}

export interface TextTheme extends TextAlignCfg {
  /* 字体 */
  fontFamily?: string;
  /* 字体大小 */
  fontSize?: number;
  /* 字体粗细 */
  fontWeight?: number | 'normal' | 'bold' | 'bolder' | 'lighter';
  /* 字体颜色 */
  fill?: string;
  /* 链接文本颜色 */
  linkTextFill?: string;
  /* 字体透明度 */
  opacity?: number;
}

export interface CellTheme {
  /* 奇数行单元格背景色 */
  crossBackgroundColor?: string;
  /* 单元格背景色 */
  backgroundColor?: string;
  /* 单元格背景色透明度 */
  backgroundColorOpacity?: number;
  /* 单元格水平边线颜色 */
  horizontalBorderColor?: string;
  /* 单元格水平边线颜色透明度 */
  horizontalBorderColorOpacity?: number;
  /* 单元格垂直边线颜色 */
  verticalBorderColor?: string;
  /* 单元格垂直边线颜色透明度 */
  verticalBorderColorOpacity?: number;
  /* 单元格水平边线宽度 */
  horizontalBorderWidth?: number;
  /* 单元格垂直边线宽度 */
  verticalBorderWidth?: number;
  /* 单元格内边距 */
  padding: Padding;
  /* 交互态 */
  interactionState?: InteractionState;
  /* 单元格内条件格式-迷你条形图高度 */
  miniBarChartHeight?: number;
  /* 单元格内条件格式-迷你条形图默认填充颜色 */
  miniBarChartFillColor?: string;
}

export interface IconTheme {
  /* icon 填充色 */
  fill?: string;
  /* 下跌 icon 填充色 */
  downIconColor?: string;
  /* 上涨 icon 填充色 */
  upIconColor?: string;
  /* icon  大小 */
  size?: number;
  /* icon 外边距 */
  margin?: Margin;
}

export interface ResizeArea {
  /* 热区尺寸 */
  size?: number;
  /* 热区背景色 */
  background?: string;
  /* 参考线颜色 */
  guideLineColor?: string;
  /* 参考线间隔 */
  guideLineDash?: number[];
  /* 热区背景色透明度 */
  backgroundOpacity?: number;
  /* 交互态 */
  interactionState?: InteractionState;
}

export interface ScrollBarTheme {
  /* 滚动条轨道颜色 */
  trackColor?: string;
  /* 滚动条 hover 态颜色 */
  thumbHoverColor?: string;
  /* 滚动条颜色 */
  thumbColor?: string;
  /* 滚动条尺寸 */
  size?: number;
  /* 滚动条 hover 态尺寸 */
  hoverSize?: number;
  /** 指定如何绘制每一条线段末端，lineCap?: 'butt' | 'round' | 'square'; */
  lineCap?: ShapeAttrs['lineCap'];
}

export interface SplitLine {
  /* 水平分割线颜色 */
  horizontalBorderColor?: string;
  /* 水平分割线颜色透明度 */
  horizontalBorderColorOpacity?: number;
  /* 水平分割线宽度 */
  horizontalBorderWidth?: number;
  /* 垂直分割线颜色 */
  verticalBorderColor?: string;
  /* 垂直分割线颜色透明度 */
  verticalBorderColorOpacity?: number;
  /* 垂直分割线宽度 */
  verticalBorderWidth?: number;
  /* 分割线是否显示外阴影 */
  showShadow?: boolean;
  /* 阴影宽度 */
  shadowWidth?: number;
  /* 阴影线性渐变色 */
  shadowColors?: {
    /* 线性变化左侧颜色 */
    left: string;
    /* 线性变化右侧颜色 */
    right: string;
  };
}
export interface DefaultCellTheme extends GridAnalysisCellTheme {
  /* 粗体文本样式 */
  bolderText?: TextTheme;
  /* 文本样式 */
  text?: TextTheme;
  /* 单元格样式 */
  cell?: CellTheme;
  /* 图标样式 */
  icon?: IconTheme;
  /* 序号列宽 */
  seriesNumberWidth?: number;
}

export interface GridAnalysisCellTheme {
  // 次级文本，如副指标
  minorText?: TextTheme;
  // 衍生指标
  derivedMeasureText?: {
    mainUp: string;
    mainDown: string;
    minorUp: string;
    minorDown: string;
  };
}

export type CellThemes = {
  [K in CellTypes]?: DefaultCellTheme;
};

export interface S2Theme extends CellThemes {
  /* 列宽行高调整热区 */
  resizeArea?: ResizeArea;
  /* 滚动条样式 */
  scrollBar?: ScrollBarTheme;
  /* 分割线样式 */
  splitLine?: SplitLine;
  /* 刷选遮罩 */
  prepareSelectMask?: InteractionStateTheme;
  /* 画布背景底色 */
  background?: Background;
}

export type ThemeName = 'default' | 'colorful' | 'gray';

export interface ThemeCfg {
  /* 主题 */
  theme?: S2Theme;
  /* 色板 */
  palette?: Palette;
  /* 主题名 */
  name?: ThemeName;
}
