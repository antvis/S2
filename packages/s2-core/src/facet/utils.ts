import { SimpleBBox, IGroup } from '@antv/g-canvas';
import { findIndex, isNil } from 'lodash';

import { Indexes } from '../utils/indexes';
import { ViewCellHeights } from './layout/interface';
import {
  FrozenCellType,
  FrozenOpts,
  FrozenCellIndex,
} from '@/common/constant/frozen';
import { Pagination, ScrollSpeedRatio } from '@/common/interface';

export const isFrozenCol = (colIndex: number, frozenCount: number) => {
  return frozenCount > 0 && colIndex < frozenCount;
};

export const isFrozenTrailingCol = (
  colIndex: number,
  frozenCount: number,
  colLength: number,
) => {
  return frozenCount > 0 && colIndex >= colLength - frozenCount;
};

export const isFrozenRow = (
  rowIndex: number,
  minRowIndex: number,
  frozenCount: number,
) => {
  return frozenCount > 0 && rowIndex < minRowIndex + frozenCount;
};

export const isFrozenTrailingRow = (
  rowIndex: number,
  maxRowIndex: number,
  frozenCount: number,
) => {
  return frozenCount > 0 && rowIndex >= maxRowIndex + 1 - frozenCount;
};

/**
 * 计算偏移 scrollX、scrollY 的时候，在视窗中的节点索引
 * @param scrollX
 * @param scrollY
 * @param widths
 * @param heights
 * @param viewport
 * @param rowRemainWidth
 */
export const calculateInViewIndexes = (
  scrollX: number,
  scrollY: number,
  widths: number[],
  heights: ViewCellHeights,
  viewport: SimpleBBox,
  rowRemainWidth?: number,
): Indexes => {
  // 1. 计算 x min、max
  let xMin = findIndex(
    widths,
    (width: number, idx: number) => {
      const x =
        scrollX - (isNil(rowRemainWidth) ? 0 : rowRemainWidth) + viewport.x;
      return x >= width && x < widths[idx + 1];
    },
    0,
  );
  xMin = Math.max(xMin, 0);

  let xMax = findIndex(
    widths,
    (width: number, idx: number) => {
      const x = viewport.width + scrollX + viewport.x;
      return x >= width && x < widths[idx + 1];
    },
    xMin,
  );
  xMax = Math.min(xMax === -1 ? Infinity : xMax, widths.length - 2);

  const { start: yMin, end: yMax } = heights.getIndexRange(
    scrollY + viewport.y,
    viewport.height + scrollY + viewport.y,
  );

  // use direction
  // const halfWidthSize = Math.ceil(xMax - xMin / 4);
  // const halfHeightSize = Math.ceil(yMax - yMin / 4);
  // xMin = Math.max(0, xMin - halfWidthSize)
  // xMax = xMax + halfWidthSize;
  // yMin = Math.max(0, yMin - halfHeightSize);
  // yMax = yMax + halfHeightSize;

  return [xMin, xMax, yMin, yMax];
};

/**
 * 优化滚动方向，对于小角度的滚动，固定为一个方向
 * @param x
 * @param y
 * @param ratio
 */
export const optimizeScrollXY = (
  x: number,
  y: number,
  ratio: ScrollSpeedRatio,
): [number, number] => {
  const ANGLE = 2; // 调参工程师
  const angle = Math.abs(x / y);

  // 经过滚动优化之后的 x, y
  const deltaX = angle <= 1 / ANGLE ? 0 : x;
  const deltaY = angle > ANGLE ? 0 : y;

  return [deltaX * ratio.horizontal, deltaY * ratio.vertical];
};

export const translateGroup = (
  group: IGroup,
  scrollX: number,
  scrollY: number,
) => {
  const matrix = group?.getMatrix();
  const preX = matrix?.[6] ?? 0;
  const preY = matrix?.[7] ?? 0;
  group?.translate(scrollX - preX, scrollY - preY);
};

export const translateGroupX = (group: IGroup, scrollX: number) => {
  const matrix = group?.getMatrix();
  const preX = matrix?.[6] ?? 0;
  group?.translate(scrollX - preX, 0);
};

export const translateGroupY = (group: IGroup, scrollY: number) => {
  const matrix = group?.getMatrix();
  const preY = matrix?.[7] ?? 0;
  group?.translate(0, scrollY - preY);
};

/**
 * frozen                     frozenTrailing
 * ColCount                   ColCount
 * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * |     |     frozenRow     |          |  frozenRowCount
 * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * | fro |                   | fro      |
 * | zen |      panel        | zen      |
 * | col |      scroll       | trailing |
 * |     |                   | col      |
 * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * |     | frozenTrailingRow |          |  frozenTrailingRowCount
 * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * @description returns which group data cell belongs in frozen mode
 */
export const getFrozenDataCellType = (
  meta: {
    colIndex: number;
    rowIndex: number;
  },
  frozenOpts: FrozenOpts,
  colLength: number,
  cellRange: {
    start: number;
    end: number;
  },
) => {
  const {
    frozenColCount,
    frozenRowCount,
    frozenTrailingColCount,
    frozenTrailingRowCount,
  } = frozenOpts;
  const { colIndex, rowIndex } = meta;
  if (isFrozenRow(rowIndex, cellRange.start, frozenRowCount)) {
    return FrozenCellType.ROW;
  }

  if (isFrozenTrailingRow(rowIndex, cellRange.end, frozenTrailingRowCount)) {
    return FrozenCellType.TRAILING_ROW;
  }

  if (isFrozenCol(colIndex, frozenColCount)) {
    return FrozenCellType.COL;
  }

  if (isFrozenTrailingCol(colIndex, frozenTrailingColCount, colLength)) {
    return FrozenCellType.TRAILING_COL;
  }
  return FrozenCellType.SCROLL;
};

/**
 * @description calculate all cells in frozen group's intersection region
 */
export const calculateFrozenCornerCells = (
  opts: FrozenOpts,
  colLength: number,
  cellRange: {
    start: number;
    end: number;
  },
) => {
  const {
    frozenColCount,
    frozenRowCount,
    frozenTrailingColCount,
    frozenTrailingRowCount,
  } = opts;

  const result: {
    [key: string]: FrozenCellIndex[];
  } = {
    [FrozenCellType.TOP]: [],
    [FrozenCellType.BOTTOM]: [],
  };

  // frozenColGroup with frozenRowGroup or frozenTrailingRowGroup. Top left and bottom left corner.
  for (let i = 0; i < frozenColCount; i++) {
    for (let j = cellRange.start; j < cellRange.start + frozenRowCount; j++) {
      result[FrozenCellType.TOP].push({
        x: i,
        y: j,
      });
    }

    if (frozenTrailingRowCount > 0) {
      for (let j = 0; j < frozenTrailingRowCount; j++) {
        const index = cellRange.end - j;
        result[FrozenCellType.BOTTOM].push({
          x: i,
          y: index,
        });
      }
    }
  }

  // frozenTrailingColGroup with frozenRowGroup or frozenTrailingRowGroup. Top right and bottom right corner.
  for (let i = 0; i < frozenTrailingColCount; i++) {
    const colIndex = colLength - 1 - i;
    for (let j = cellRange.start; j < cellRange.start + frozenRowCount; j++) {
      result[FrozenCellType.TOP].push({
        x: colIndex,
        y: j,
      });
    }

    if (frozenTrailingRowCount > 0) {
      for (let j = 0; j < frozenTrailingRowCount; j++) {
        const index = cellRange.end - j;
        result[FrozenCellType.BOTTOM].push({
          x: colIndex,
          y: index,
        });
      }
    }
  }

  return result;
};

export const isFrozenCell = (
  colIndex: number,
  rowIndex: number,
  frozenOpts: FrozenOpts,
  colLength: number,
  cellRange: {
    start: number;
    end: number;
  },
) => {
  const {
    frozenColCount,
    frozenRowCount,
    frozenTrailingColCount,
    frozenTrailingRowCount,
  } = frozenOpts;
  return (
    isFrozenCol(colIndex, frozenColCount) ||
    isFrozenTrailingCol(colIndex, frozenTrailingColCount, colLength) ||
    isFrozenRow(rowIndex, cellRange.start, frozenRowCount) ||
    isFrozenTrailingRow(rowIndex, cellRange.end, frozenTrailingRowCount)
  );
};

/**
 * @description split all cells in current panel with five child group
 */
export const splitInViewIndexesWithFrozen = (
  indexes: Indexes,
  frozenOpts: FrozenOpts,
  colLength: number,
  cellRange: {
    start: number;
    end: number;
  },
) => {
  const {
    frozenColCount,
    frozenRowCount,
    frozenTrailingColCount,
    frozenTrailingRowCount,
  } = frozenOpts;

  const centerIndexes: Indexes = [...indexes];

  // Cut off frozen cells from centerIndexes
  if (isFrozenCol(centerIndexes[0], frozenColCount)) {
    centerIndexes[0] = frozenColCount;
  }

  if (
    isFrozenTrailingCol(centerIndexes[1], frozenTrailingColCount, colLength)
  ) {
    centerIndexes[1] = colLength - frozenTrailingColCount - 1;
  }

  if (isFrozenRow(centerIndexes[2], cellRange.start, frozenRowCount)) {
    centerIndexes[2] = cellRange.start + frozenRowCount;
  }

  if (
    isFrozenTrailingRow(centerIndexes[3], cellRange.end, frozenTrailingRowCount)
  ) {
    centerIndexes[3] = cellRange.end - frozenTrailingRowCount;
  }

  // Calculate indexes for four frozen groups
  const frozenRowIndexes: Indexes = [...centerIndexes];
  frozenRowIndexes[2] = cellRange.start;
  frozenRowIndexes[3] = cellRange.start + frozenRowCount - 1;

  const frozenColIndexes: Indexes = [...centerIndexes];
  frozenColIndexes[0] = 0;
  frozenColIndexes[1] = frozenColCount - 1;

  const frozenTrailingRowIndexes: Indexes = [...centerIndexes];
  frozenTrailingRowIndexes[2] = cellRange.end + 1 - frozenTrailingRowCount;
  frozenTrailingRowIndexes[3] = cellRange.end;

  const frozenTrailingColIndexes: Indexes = [...centerIndexes];
  frozenTrailingColIndexes[0] = colLength - frozenTrailingColCount;
  frozenTrailingColIndexes[1] = colLength - 1;

  return {
    center: centerIndexes,
    frozenRow: frozenRowIndexes,
    frozenCol: frozenColIndexes,
    frozenTrailingCol: frozenTrailingColIndexes,
    frozenTrailingRow: frozenTrailingRowIndexes,
  };
};

export const getCellRange = (
  viewCellHeights: ViewCellHeights,
  pagination?: Pagination,
) => {
  const heights = viewCellHeights;
  let start = 0;
  let end = heights.getTotalLength() - 1;

  if (pagination) {
    const { current, pageSize } = pagination;

    start = Math.max((current - 1) * pageSize, 0);
    end = Math.min(current * pageSize - 1, heights.getTotalLength() - 1);
  }

  return {
    start,
    end,
  };
};
