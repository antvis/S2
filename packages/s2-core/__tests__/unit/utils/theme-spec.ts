import { getPalette } from '../../../src/utils/theme';

describe('Theme Utils Tests', () => {
  test('should get correctly palette', () => {
    expect(getPalette()).toMatchSnapshot();
    expect(getPalette('default')).toMatchSnapshot();
    expect(getPalette('colorful')).toMatchSnapshot();
    expect(getPalette('gray')).toMatchSnapshot();
  });
});
