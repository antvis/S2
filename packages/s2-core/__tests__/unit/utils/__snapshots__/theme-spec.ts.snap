// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Utils Tests should get correctly palette 1`] = `
Object {
  "basicColorRelations": Array [
    Object {
      "basicColorIndex": 1,
      "standardColorIndex": 0,
    },
    Object {
      "basicColorIndex": 2,
      "standardColorIndex": 1,
    },
    Object {
      "basicColorIndex": 3,
      "standardColorIndex": 1,
    },
    Object {
      "basicColorIndex": 4,
      "standardColorIndex": 2,
    },
    Object {
      "basicColorIndex": 5,
      "standardColorIndex": 7,
    },
    Object {
      "basicColorIndex": 6,
      "standardColorIndex": 5,
    },
    Object {
      "basicColorIndex": 7,
      "standardColorIndex": 5,
    },
    Object {
      "basicColorIndex": 9,
      "standardColorIndex": 1,
    },
    Object {
      "basicColorIndex": 10,
      "standardColorIndex": 2,
    },
    Object {
      "basicColorIndex": 11,
      "standardColorIndex": 5,
    },
    Object {
      "basicColorIndex": 12,
      "standardColorIndex": 5,
    },
  ],
  "basicColors": Array [
    "#000000",
    "#F5F8FE",
    "#E0E9FD",
    "#E0E9FD",
    "#CCDBFC",
    "#234DAB",
    "#326EF4",
    "#326EF4",
    "#FFFFFF",
    "#E0E9FD",
    "#CCDBFC",
    "#326EF4",
    "#326EF4",
    "#000000",
    "#000000",
  ],
  "brandColor": "#326EF4",
  "others": Object {
    "highlight": "#87B5FF",
    "results": "#F0F7FF",
  },
  "semanticColors": Object {
    "green": "#29A294",
    "red": "#FF4D4F",
    "yellow": "#FAAD14",
  },
}
`;

exports[`Theme Utils Tests should get correctly palette 2`] = `
Object {
  "basicColorRelations": Array [
    Object {
      "basicColorIndex": 1,
      "standardColorIndex": 0,
    },
    Object {
      "basicColorIndex": 2,
      "standardColorIndex": 1,
    },
    Object {
      "basicColorIndex": 3,
      "standardColorIndex": 1,
    },
    Object {
      "basicColorIndex": 4,
      "standardColorIndex": 2,
    },
    Object {
      "basicColorIndex": 5,
      "standardColorIndex": 7,
    },
    Object {
      "basicColorIndex": 6,
      "standardColorIndex": 5,
    },
    Object {
      "basicColorIndex": 7,
      "standardColorIndex": 5,
    },
    Object {
      "basicColorIndex": 9,
      "standardColorIndex": 1,
    },
    Object {
      "basicColorIndex": 10,
      "standardColorIndex": 2,
    },
    Object {
      "basicColorIndex": 11,
      "standardColorIndex": 5,
    },
    Object {
      "basicColorIndex": 12,
      "standardColorIndex": 5,
    },
  ],
  "basicColors": Array [
    "#000000",
    "#F5F8FE",
    "#E0E9FD",
    "#E0E9FD",
    "#CCDBFC",
    "#234DAB",
    "#326EF4",
    "#326EF4",
    "#FFFFFF",
    "#E0E9FD",
    "#CCDBFC",
    "#326EF4",
    "#326EF4",
    "#000000",
    "#000000",
  ],
  "brandColor": "#326EF4",
  "others": Object {
    "highlight": "#87B5FF",
    "results": "#F0F7FF",
  },
  "semanticColors": Object {
    "green": "#29A294",
    "red": "#FF4D4F",
    "yellow": "#FAAD14",
  },
}
`;

exports[`Theme Utils Tests should get correctly palette 3`] = `
Object {
  "basicColorRelations": Array [
    Object {
      "basicColorIndex": 1,
      "standardColorIndex": 0,
    },
    Object {
      "basicColorIndex": 2,
      "standardColorIndex": 1,
    },
    Object {
      "basicColorIndex": 3,
      "standardColorIndex": 5,
    },
    Object {
      "basicColorIndex": 4,
      "standardColorIndex": 6,
    },
    Object {
      "basicColorIndex": 5,
      "standardColorIndex": 6,
    },
    Object {
      "basicColorIndex": 6,
      "standardColorIndex": 6,
    },
    Object {
      "basicColorIndex": 7,
      "standardColorIndex": 5,
    },
    Object {
      "basicColorIndex": 9,
      "standardColorIndex": 1,
    },
    Object {
      "basicColorIndex": 10,
      "standardColorIndex": 4,
    },
    Object {
      "basicColorIndex": 11,
      "standardColorIndex": 4,
    },
    Object {
      "basicColorIndex": 12,
      "standardColorIndex": 5,
    },
  ],
  "basicColors": Array [
    "#FFFFFF",
    "#F5F8FF",
    "#E1EAFE",
    "#3471F9",
    "#2C60D4",
    "#2C60D4",
    "#2C60D4",
    "#3471F9",
    "#FFFFFF",
    "#E1EAFE",
    "#5286FA",
    "#5286FA",
    "#3471F9",
    "#000000",
    "#000000",
  ],
  "brandColor": "#3471F9",
  "others": Object {
    "highlight": "#87B5FF",
    "results": "#F0F7FF",
  },
  "semanticColors": Object {
    "green": "#29A294",
    "red": "#FF4D4F",
    "yellow": "#FAAD14",
  },
}
`;

exports[`Theme Utils Tests should get correctly palette 4`] = `
Object {
  "basicColorRelations": Array [
    Object {
      "basicColorIndex": 1,
      "standardColorIndex": 0,
    },
    Object {
      "basicColorIndex": 2,
      "standardColorIndex": 1,
    },
    Object {
      "basicColorIndex": 3,
      "standardColorIndex": 1,
    },
    Object {
      "basicColorIndex": 4,
      "standardColorIndex": 2,
    },
    Object {
      "basicColorIndex": 5,
      "standardColorIndex": 7,
    },
    Object {
      "basicColorIndex": 6,
      "standardColorIndex": 8,
    },
    Object {
      "basicColorIndex": 7,
      "standardColorIndex": 5,
    },
    Object {
      "basicColorIndex": 9,
      "standardColorIndex": 1,
    },
    Object {
      "basicColorIndex": 10,
      "standardColorIndex": 2,
    },
    Object {
      "basicColorIndex": 11,
      "standardColorIndex": 3,
    },
    Object {
      "basicColorIndex": 12,
      "standardColorIndex": 3,
    },
  ],
  "basicColors": Array [
    "#000000",
    "#FAFBFB",
    "#F0F2F4",
    "#F0F2F4",
    "#E7E9ED",
    "#6E757F",
    "#565C64",
    "#9DA7B6",
    "#FFFFFF",
    "#F0F2F4",
    "#E7E9ED",
    "#BAC1CC",
    "#BAC1CC",
    "#000000",
    "#000000",
  ],
  "brandColor": "#9DA7B6",
  "others": Object {
    "highlight": "#87B5FF",
    "results": "#F0F7FF",
  },
  "semanticColors": Object {
    "green": "#29A294",
    "red": "#FF4D4F",
    "yellow": "#FAAD14",
  },
}
`;
